File: 프로그래머스/unrated/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.py
def solution(sequence, k):
    sequence.append(int(1e6))
    n = len(sequence)
    
    interval_sum = 0
    s, e = 0, 0
    k_seq = []
    
    while e < n:
        if interval_sum >= k:
            interval_sum -= sequence[s]
            s += 1
        elif interval_sum < k:
            interval_sum += sequence[e]
            e += 1
        
        if interval_sum == k:
            k_seq.append((s, e - 1, e - s))
    
    k_seq.sort(key = lambda x : int(x[2]))
    return [k_seq[0][0], k_seq[0][1]]

File: 프로그래머스/unrated/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.java
import java.util.*;

class Solution {
    public int[] solution(int[] sequence, int k) {
        sequence = Arrays.copyOf(sequence, sequence.length + 1);
        sequence[sequence.length - 1] = (int) 1e6;
        int start = 0, end = 0;
        int pSum = 0;
        ArrayList<int[]> result = new ArrayList<>(); 
        
        while (end < sequence.length) {
            if (pSum >= k) {
                pSum -= sequence[start];
                start++;
            } else if (pSum < k) {
                pSum += sequence[end];
                end++;
            }
            if (pSum == k) {
                result.add(new int[]{start, end - 1, end - start});
            }
        }
        // for (int[] intArr : result) {
        //     System.out.println(Arrays.toString(intArr));
        // }
        Collections.sort(result, (o1, o2) -> o1[2] - o2[2]);
        
        return new int[]{result.get(0)[0], result.get(0)[1]};
    }
}